sunpy/net/hek/hek.py

```diff
@@ -54,11 +54,12 @@ class HEKClient(object):
     def __init__(self, url=DEFAULT_URL):
         self.url = url
 
-    def _download(self, data):
+    def _download(self, data, select=None):
         """ Download all data, even if pagiated. """
         page = 1
         results = []
-
+        if select is not None and type(select) is list:
```

Comment:
drop `type(select) is list`. Leads to confusing behaviour. Also could be any iterable anyway.